### PR TITLE
Add rating for lessons

### PR DESCRIPTION
**Pull request by PandaGrom**

### CODE REVIEW

File: app/controllers/lessons_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-class LessonsController < ApplicationController
+class LessonsController < AuthorizationController
```
Comment (fooooxmr) [inline app/controllers/lessons_controller.rb:3]: Lessons endpoint should be available for non registred users.

File: app/controllers/lessons_controller.rb [hunk 2]
```diff
@@ -35,6 +35,7 @@ def create
   def edit; end
 
   def update
+    CalculateLessonsRating.new(lesson: @lesson, current_user:, rating: lesson_rating_params[:rating]).call
```
Comment (fooooxmr) [inline app/controllers/lessons_controller.rb:38]: Why you are passing current_user: nil?

File: app/models/lesson.rb [hunk 3]
```diff
@@ -21,6 +22,7 @@ class Lesson < ApplicationRecord
                                     message: 'This character is not available for input in this field' }
   validates :video_link, presence: true, url: { message: 'Please check the correctness of the link' }
   validates :category_id, presence: true
+  validates :rating, numericality: { less_than_or_equal_to: 5 }
```
Comment (fooooxmr) [inline app/models/lesson.rb:25]: Rating can not be less then 0.

File: app/services/calculate_lessons_rating.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+class CalculateLessonsRating
```
Comment (fooooxmr) [inline app/services/calculate_lessons_rating.rb:1]: It's better to name CalculateLessonsRatingService to make understandable, that this is a service.

File: app/services/calculate_lessons_rating.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  def create_lesson_rating
+    LessonRating.create!(user_id: @current_user.id, lesson_id: @lesson.id, rating: @rating) if create?
+  end
+
+  def update_lesson_rating
+    LessonRating.find_by(user_id: @current_user.id, lesson_id: @lesson.id).update!(rating: @rating)
+  end
+
+  def calculate_lesson_rating
+    @lesson.lesson_ratings.average(:rating)
```
Comment (fooooxmr) [inline app/services/calculate_lessons_rating.rb:32]: What is happening in this method?

File: spec/models/lesson_rating_spec.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (fooooxmr) [inline spec/models/lesson_rating_spec.rb:1]: Do not forget to update swagger.

File: app/controllers/lessons_controller.rb (hunk 1 above)
Comment (fooooxmr) [reply app/controllers/lessons_controller.rb:3]: Only endpoint with setting rating should be secure from guests.

**Review by fooooxmr — CHANGES_REQUESTED** [review]

**Review by evrusha — COMMENTED** [review]
I will re-review after conversations be resolved.

**Review by Terapool — APPROVED** [review]

**Review by fooooxmr — APPROVED** [review]

**Review by evrusha — APPROVED** [review]

**Pull request merged**